tools/Customization/DevHome.Customization/ViewModels/DevDriveInsights/OptimizeDevDriveDialogViewModel.cs

```diff
@@ -86,6 +98,18 @@ private async Task BrowseButtonClick(object sender)
         if (folder != null)
         {
             DirectoryPathTextBox = folder.Path;
+            if (ChosenDirectoryInDevDrive(DirectoryPathTextBox))
+            {
+                IsPrimaryButtonEnabled = true;

```

Comment:
Hmm you can probably use the CanExecute decoration over your DirectoryInputConfirmedCommand method. example:

https://github.com/microsoft/devhome/blob/e190cddf9141b3ee31ddcbb0a6c6b6b511300ab6/tools/SetupFlow/DevHome.SetupFlow/ViewModels/DevDriveViewModel.cs#L138


e.g 
```C#
    [ObservableProperty]
    [NotifyCanExecuteChangedFor(nameof(DirectoryInputConfirmedCommand))]
    private string _errorMessage;

    [ObservableProperty]
    [NotifyCanExecuteChangedFor(nameof(DirectoryInputConfirmedCommand))]
    private string _directoryPathTextBox;

    // Then for your DirectoryInputConfirmed relay command
    [RelayCommand(CanExecute = nameof(DirectoryInfoConfirmed))]
    private void DirectoryInputConfirmed()
    {
        // ...
    }

    // Called when ErrorMessage's value is changed
    private bool DirectoryInfoConfirmed()
    {
        return string.IsNullOrEmpty(ErrorMessage)  && !string.IsNullOrEmpty(DirectoryPathTextBox);
    }
```

What that will do is every time `ErrorMessage` or `DirectoryPathTextBox` are updated, it will cause  a check to happen for the `DirectoryInputConfirmed` relay command. Which will run `DirectoryInfoConfirmed()`. If `DirectoryInfoConfirmed()` returns true then the primary button will be enabled. If its false then it will stay false. This way you won't need the `IsPrimaryButtonEnabled` I believe